#D1. Creating the price_label column.

In [4]:
#import df orders_products_merged 
import pandas as pd
import numpy as np
import os  

path= r'/Users/katep/08-2023 Instacart Basket Analysis'
df_ords_prods_merged = pd.read_pickle(os.path.join(path, '02 Data', 'Prepared Data', 'orders_products_merged.pkl'))

df_ords_prods_merged.columns

Index(['Unnamed: 0_x', 'order_id', 'user_id', 'order_sequence',
       'orders_day_of_week', 'order_hour_of_day', 'days_since_prior_order',
       'product_id', 'add_to_cart_order', 'reordered', 'Unnamed: 0_y',
       'product_name', 'aisle_id', 'department_id', 'prices', '_merge'],
      dtype='object')

In [7]:
#Create a new column 'price_range_loc' in df_ords_prods_merged
df_ords_prods_merged.loc[df_ords_prods_merged['prices'] > 15, 'price_range'] = 'High-range product'
df_ords_prods_merged.loc[(df_ords_prods_merged['prices'] <= 15) & (df_ords_prods_merged['prices'] > 5), 'price_range'] = 'Mid-range product' 
df_ords_prods_merged.loc[df_ords_prods_merged['prices'] <= 5, 'price_range'] = 'Low-range product'

In [8]:
df_ords_prods_merged['price_range'].value_counts(dropna =False)

price_range
Mid-range product     21889009
Low-range product     10126339
High-range product      417682
Name: count, dtype: int64

#D1. Creating the busiest_day column.

In [9]:
df_ords_prods_merged['orders_day_of_week'].value_counts(dropna =False)

orders_day_of_week
0    6209410
1    5665604
6    4500101
2    4217610
5    4209334
3    3843929
4    3787042
Name: count, dtype: int64

In [12]:
result = []

for value in df_ords_prods_merged["orders_day_of_week"]:
  if value == 0:
    result.append("Busiest day")
  elif value == 4:
    result.append("Least busy")
  else:
    result.append("Regularly busy")
    

In [13]:
df_ords_prods_merged['busiest_day'] = result

In [14]:
df_ords_prods_merged['busiest_day'].value_counts(dropna = False)

busiest_day
Regularly busy    22436578
Busiest day        6209410
Least busy         3787042
Name: count, dtype: int64

#D2. Create a column for 2 Busiest days and 2 slowest days.

In [15]:
result2 = []

for value in df_ords_prods_merged["orders_day_of_week"]:
  if (value == 0)or (value==1):
    result2.append("Busiest days")
  elif (value == 3)or (value ==4):
    result2.append("Slowest days")
  else:
    result2.append("Regularly busy")
    

In [16]:
df_ords_prods_merged['Busiest_days'] = result2

In [17]:
#D3. Check the values in the new column.
df_ords_prods_merged['Busiest_days'].value_counts(dropna = False)

Busiest_days
Regularly busy    12927045
Busiest days      11875014
Slowest days       7630971
Name: count, dtype: int64

#D3. Busiest days  11875014 = orders_day_of_week (0, 6209410) + (1, 5665604)
     Slowest days  7630971  = orders_day_of_week (3, 3843929) + (4, 3787042)

In [18]:
df_ords_prods_merged.columns

Index(['Unnamed: 0_x', 'order_id', 'user_id', 'order_sequence',
       'orders_day_of_week', 'order_hour_of_day', 'days_since_prior_order',
       'product_id', 'add_to_cart_order', 'reordered', 'Unnamed: 0_y',
       'product_name', 'aisle_id', 'department_id', 'prices', '_merge',
       'price_range', 'busiest_day', 'Busiest_days'],
      dtype='object')

In [19]:
#Check the order frequency of the hour
df_ords_prods_merged['order_hour_of_day'].value_counts(dropna =False)

order_hour_of_day
10    2764288
11    2738483
14    2691448
15    2664420
13    2663169
12    2620719
16    2537358
9     2456591
17    2089385
8     1719888
18    1637858
19    1259335
20     976991
7      891900
21     796341
22     634715
23     402593
6      290763
0      218925
1      115780
5       88054
2       69429
4       53280
3       51317
Name: count, dtype: int64

In [19]:
#D4. Create a new column containing these labels called “busiest_period_of_day.”
df_ords_prods_merged.loc[(df_ords_prods_merged['order_hour_of_day'] > 8)&(df_ords_prods_merged['order_hour_of_day'] < 17), 'busiest_period_of_day'] = 'Most orders'
df_ords_prods_merged.loc[(df_ords_prods_merged['order_hour_of_day'] >= 17)&(df_ords_prods_merged['order_hour_of_day'] < 23), 'busiest_period_of_day'] = 'Average orders' 
df_ords_prods_merged.loc[(df_ords_prods_merged['order_hour_of_day'] >= 7)&(df_ords_prods_merged['order_hour_of_day'] < 9), 'busiest_period_of_day'] = 'Average orders' 
df_ords_prods_merged.loc[(df_ords_prods_merged['order_hour_of_day'] >= 0)&(df_ords_prods_merged['order_hour_of_day'] < 7), 'busiest_period_of_day'] = 'Fewest orders'
df_ords_prods_merged.loc[df_ords_prods_merged['order_hour_of_day']==23, 'busiest_period_of_day'] = 'Fewest orders'


In [20]:
#D5. Print the frequency for this new column.
df_ords_prods_merged['busiest_period_of_day'].value_counts(dropna =False)

busiest_period_of_day
Most orders       21136476
Average orders    10006413
Fewest orders      1290141
Name: count, dtype: int64

In [21]:
df_ords_prods_merged.columns

Index(['Unnamed: 0_x', 'order_id', 'user_id', 'order_sequence',
       'orders_day_of_week', 'order_hour_of_day', 'days_since_prior_order',
       'product_id', 'add_to_cart_order', 'reordered', 'Unnamed: 0_y',
       'product_name', 'aisle_id', 'department_id', 'prices', '_merge',
       'price_range', 'busiest_day', 'Busiest_days', 'busiest_period_of_day'],
      dtype='object')

In [22]:
#D7.Export the newly created dataframe as orders_products_merged2.
df_ords_prods_merged.to_pickle(os.path.join(path, '02 Data','Prepared Data', 'orders_products_merged2.pkl'))